src/dbnode/storage/shard.go

```diff
@@ -2999,14 +3009,18 @@ func (s *dbShard) logFlushResult(r dbShardFlushResult) {
 	)
 }
 
-func (s *dbShard) finishWriting(startTime time.Time, nextVersion int) error {
+func (s *dbShard) finishWriting(
+	startTime time.Time,
+	nextVersion int,
+	setInitialFileOpStatus bool,
+) error {
 	// After writing the full block successfully update the ColdVersionFlushed number. This will
 	// allow the SeekerManager to open a lease on the latest version of the fileset files because
 	// the BlockLeaseVerifier will check the ColdVersionFlushed value, but the buffer only looks at
 	// ColdVersionRetrievable so a concurrent tick will not yet cause the blocks in memory to be
 	// evicted (which is the desired behavior because we haven't updated the open leases yet which
 	// means the newly written data is not available for querying via the SeekerManager yet.)
-	s.setFlushStateColdVersionFlushed(startTime, nextVersion)
+	s.setFlushStateColdVersionFlushed(startTime, nextVersion, setInitialFileOpStatus)
```

Comment:
I think it would be cleaner to directly call `markWarmFlushStateSuccess()` instead of changing `setFlushStateColdVersionFlushed()` since it's doing more work than the fn name suggests now.

Reply:
Good point, changing.